Framework/Geometry/src/Instrument/InstrumentVisitor.cpp

```diff
@@ -203,6 +205,7 @@ InstrumentVisitor::registerGenericComponent(const IComponent &component) {
   // Unless this is the root component this parent is not correct and will be
   // updated later in the register call of the parent.
   m_parentComponentIndices->push_back(componentIndex);
+  m_children->emplace_back(std::vector<size_t>());
```

Comment:
Code comment above to the effect that generic components are not assemblies and cannot have children.